# HG changeset patch
# User Konstantinos Poulios <logari81@googlemail.com>
# Date 1295947803 -3600
#      Tue Jan 25 10:30:03 2011 +0100
# Branch release-3-4-x
# Node ID 5356c44ecede997342452bfb5655d4b245e32779
# Parent  e6295ea9ac8ad2284c7e895719dec218c2030949
fix wrong dimensions of minor ticks vector

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-25  Konstantinos Poulios  <logari81@googlemail.com>
+
+	* src/graphics.cc (axes::properties::calc_ticks_and_lims):
+	Fixing wrong dimensions of minor ticks vector.
+
 2011-01-25  John W. Eaton  <jwe@octave.org>
 
 	* oct-map.h (octave_map::octave_map (const string_vector&)):
 	Initialize xvals elements to 1x1 Cells.  Bug #32222.
 
 2011-01-24  Pascal Dupuis  <Pascal.Dupuis@worldonline.be>
 
 	* ov-base-mat.h (class octave_base_matrix): Disallow assignment.
diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -4430,17 +4430,17 @@ axes::properties::calc_ticks_and_lims (a
       tmp_ticks (i) = tick_sep * (i+i1);
       if (is_logscale)
         tmp_ticks (i) = std::pow (10., tmp_ticks (i));
     }
 
   ticks = tmp_ticks;
 
   int n = is_logscale ? 9 : 4;
-  Matrix tmp_mticks (1, n * tmp_ticks.numel ());
+  Matrix tmp_mticks (1, n * (tmp_ticks.numel () - 1));
 
   for (int i = 0; i < tmp_ticks.numel ()-1; i++)
     {
       double d = (tmp_ticks (i+1) - tmp_ticks (i)) / (n+1);
       for (int j = 0; j < n; j++)
         {
           tmp_mticks (n*i+j) = tmp_ticks (i) + d * (j+1);
         }
